--- a/dlls/kernelbase/thread.c
+++ b/dlls/kernelbase/thread.c
@@ -1151,12 +1151,15 @@ BOOL WINAPI DECLSPEC_HOTPATCH ConvertFiberToThread(void)
 {
     struct fiber_data *fiber = NtCurrentTeb()->Tib.FiberData;
 
-    if (fiber)
+    if (!NtCurrentTeb()->HasFiberData)
     {
-        relocate_thread_actctx_stack( &NtCurrentTeb()->ActivationContextStack );
-        NtCurrentTeb()->Tib.FiberData = NULL;
-        HeapFree( GetProcessHeap(), 0, fiber );
+        SetLastError( ERROR_ALREADY_THREAD );
+        return FALSE;
     }
+    relocate_thread_actctx_stack( &NtCurrentTeb()->ActivationContextStack );
+    NtCurrentTeb()->Tib.FiberData = NULL;
+    NtCurrentTeb()->HasFiberData = FALSE;
+    HeapFree( GetProcessHeap(), 0, fiber );
     return TRUE;
 }
 
@@ -1181,7 +1184,7 @@ LPVOID WINAPI DECLSPEC_HOTPATCH ConvertThreadToFiberEx( LPVOID param, DWORD flags )
 {
     struct fiber_data *fiber;
 
-    if (NtCurrentTeb()->Tib.FiberData)
+    if (NtCurrentTeb()->HasFiberData)
     {
         /* CrossOver hack for bug 20987 */
         static int is_halo_mcc = -1;
@@ -1231,6 +1234,7 @@ LPVOID WINAPI DECLSPEC_HOTPATCH ConvertThreadToFiberEx( LPVOID param, DWORD flags )
     fiber->fls_slots        = NtCurrentTeb()->FlsSlots;
     relocate_thread_actctx_stack( &fiber->actctx.stack_space );
     NtCurrentTeb()->Tib.FiberData = fiber;
+    NtCurrentTeb()->HasFiberData = TRUE;
     return fiber;
 }
 
@@ -1250,7 +1254,7 @@ void WINAPI DECLSPEC_HOTPATCH DeleteFiber( LPVOID fiber_ptr )
  */
 BOOL WINAPI DECLSPEC_HOTPATCH IsThreadAFiber(void)
 {
-    return NtCurrentTeb()->Tib.FiberData != NULL;
+    return NtCurrentTeb()->HasFiberData;
 }
 
 
